plugin/distribute.go

```diff
@@ -0,0 +1,124 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+)
+
+type Distrubute struct {
+	Active       bool
+	SyncInterval time.Duration
+	lastSync     int64
+}
+
+func (e *Ratelimit) syncService() {
+	ticker := time.NewTicker(e.Distributed.SyncInterval)
+
+	syncFunc := func() error {
+		//read from the redis
+		fmt.Println("Syncing with Redis")
+
+		// currentTimeMilli := time.Now().UnixMilli()
+		//Obtain Redis lock
+		ctx := context.Background()
+
+		redisLock, err := ObtainLock(ctx, e.UniqueKey)
+		if err != nil {
+			return err
+		}
+		// fmt.Println("LOCK OBTAIN", time.Now().UnixMilli()-currentTimeMilli)
+
+		result, err := GetData(ctx, e.UniqueKey)
+		if err != nil && err != redis.Nil {
+			return fmt.Errorf("error while fetching from redis: %v", err.Error())
+		}
+		// fmt.Println("GET", time.Now().UnixMilli()-currentTimeMilli)
+
+		currentTime := time.Now().Unix()
+		syncedZones := Zones{}
+
+		// if key does not exist-> there is nothing to marshal
+		if err != redis.Nil {
+			err = json.Unmarshal([]byte(result), &syncedZones)
```

Comment:
Worth to check if json is the best serialization, standard golang serializer uses reflection, check others like https://github.com/bytedance/sonic

Reply:
shifted to sonic. I did look into others like jsoniter, but according to benchmarks it was pretty good so we are using this.